semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -35,32 +45,46 @@ object Main {
       failed = true
     }
     val classpath = Classpath(settings.cps.mkString(File.pathSeparator))
+    val scopes = new TypeVariableScopes()
     classpath.visit { root =>
       new FileVisitor[Path] {
         // Convert a .class file to a .class.semanticdb file with symbols only.
         override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
+          FileVisitResult.CONTINUE
+        }
+
+        def handleFile(file: Path): FileVisitResult = {
           if (PathIO.extension(file) != "class") return FileVisitResult.CONTINUE
           try {
             val relpath = AbsolutePath(file).toRelative(root).toString
-            val bytecode = ByteCode(Files.readAllBytes(file))
+            val bytes = Files.readAllBytes(file)
+            val bytecode = ByteCode(bytes)
             val classfile = ClassFileParser.parse(bytecode)
-            ScalaSigParser.parse(classfile) match {
-              case Some(scalaSig) =>
-                val className = NameTransformer.decode(PathIO.toUnix(relpath))
-                val semanticdbRelpath = relpath + ".semanticdb"
-                val semanticdbAbspath = semanticdbRoot.resolve(semanticdbRelpath)
-                val semanticdbDocument = s.TextDocument(
-                  schema = s.Schema.SEMANTICDB3,
-                  uri = className,
-                  language = Some(s.Language("Scala")),
-                  symbols = scalaSigPackages(scalaSig) ++ scalaSigSymbols(scalaSig))
-                val semanticdbDocuments = s.TextDocuments(List(semanticdbDocument))
-                FileIO.write(semanticdbAbspath, semanticdbDocuments)
-              case None =>
-                // NOTE: If a classfile doesn't contain ScalaSignature,
-                // we skip it for the time being. In the future, we may add support
-                // for parsing arbitrary Java classfiles.
-                ()
+
+            val isScalaFile =
+              classfile.attribute("ScalaSig").isDefined ||
+                classfile.attribute("ScalaInlineInfo").isDefined
```

Comment:
Now that we're depending ASM, how about we use ASM to figure out whether we're looking at a Scala file? I have more trust in ASM that in scalap, which isn't used on any critical path of scalac tests.